src/root/rmem.d

```diff
@@ -127,7 +127,7 @@ else
     __gshared size_t heapleft = 0;
     __gshared void* heapp;
 
-    extern (C++) void* allocmemory(size_t m_size)
+    extern (C) void* allocmemory(size_t m_size)
```

Comment:
>I would prefer you changed the C++ side to use dsize_t instead.
>>Not another name debate, sigh.
>What are you talking about?

The endless debates over what to name things, which wind up consuming thousands of message postings, and delay good PRs for months.

> Instead of changing the linkage, do what you did for every other mangling problem in this PR.

I chose to avoid using d_size_t as much as reasonable. Using C linkage is hardly unusual, it is used extensively in druntime to avoid name mangling issues. I really do not want to get into some new convention of using d_size_t.

> There might be some in libraries that other backend projects use (but probably their authors know that there is a reason to respect this common C convention, so you might be lucky).

Using a namespace prefix doesn't guarantee no conflicts, not even with C++. Dealing with a name conflict happens with C, and we just deal with it when it does. It's not some disaster, and not any more of a disaster than any of the other numerous names the dmd front end exposes.